components/ProjectDetails/index.js

```diff
@@ -1,59 +1,208 @@
-      />
-      <p>{result.description}</p>
-
-      <strong>Duration:</strong>
-      {result.duration}
-      <strong> Difficulty:</strong>
-      {result.difficulty}
-
-      {result.material && (
-        <ul>
-          <h3>Material</h3>
-          {result.material.map((material) => (
-            <li key={uuidv4()}>
-              {material.amount} {material.material}
-            </li>
-          ))}
-        </ul>
-      )}
-
-      {result.instructions && (
-        <ul>
+
+      {editMode ? (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <label>
+            Image:
+            <input {...register("image")} defaultValue={project.image} />
+          </label>
+        </form>
+      ) : (
+        <>
+          <Image
+            src={project.image}
+            height={200}
+            width={400}
+            alt={project.title}
+          />
+
+          <FavouriteButton
+            onToggleFavourite={onToggleFavourite}
+            id={project._id}
+            favourites={favourites}
+            isFavourite={isFavourite}
+          />
+        </>
+      )}
+
+      {editMode ? (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <label>
+            Title:
+            <input {...register("title")} defaultValue={project.title} />
+          </label>
+        </form>
+      ) : (
+        <h3>{project.title}</h3>
+      )}
+
+      {editMode ? (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <label>
+            Description:
+            <textarea
+              {...register("description")}
+              defaultValue={project.description}
+            />
+          </label>
+        </form>
+      ) : (
+        <p>{project.description}</p>
+      )}
+
+      {editMode ? (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <label>
+            Duration:
+            <select {...register("duration")} defaultValue={project.duration}>
+              <option value="select...">select...</option>
+              <option value="short">short</option>
+              <option value="medium">medium</option>
+              <option value="long">long</option>
+            </select>
+          </label>
+        </form>
+      ) : (
+        <>
+          <strong>Duration:</strong>
+          {project.duration}
+        </>
+      )}
+
+      {editMode ? (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <label>
+            Difficulty:
+            <select
+              {...register("difficulty")}
+              defaultValue={project.difficulty}
+            >
+              <option value="select...">select...</option>
+              <option value="easy">easy</option>
+              <option value="medium">medium</option>
+              <option value="hard">hard</option>
+            </select>
+          </label>
+        </form>
+      ) : (
+        <>
+          <strong> Difficulty:</strong>
+          {project.difficulty}
+        </>
+      )}
```

Comment:
I like the idea to make every entry editable by itself. But the singular isEditMode boolean does not really account for that. Further, you need to make multiple requests if a user wants to change multiple elements at a time. A single form where all elements from your project can be edited at once and then be updated in a single "PUT" request ist the more efficient option here. Also, you need less code to achieve this. Please update your PR to have only one Form and not one for each input individually.

Reply:
Thank you for your comment, we updated the code and added some components and only used one form now :)